base/services/dcomlaunch/network.c

```diff
@@ -104,8 +105,14 @@ CookupNodeId(UCHAR * NodeId)
         *NodeMiddle = *NodeMiddle ^ Luid.HighPart;
     }
 
+    /* Get system drive */
+    if (GetEnvironmentVariableW(L"SystemDrive", szDrive, ARRAYSIZE(szDrive) - 1))
+        lstrcatW(szDrive, L"\\");
+    else
+        lstrcpyW(szDrive, L"C:\\");
+
     /* And finally with free disk space */
-    if (GetDiskFreeSpaceA("c:\\", &SectorsPerCluster, &BytesPerSector, &NumberOfFreeClusters, &TotalNumberOfClusters))
+    if (GetDiskFreeSpaceW(szDrive, &SectorsPerCluster, &BytesPerSector, &NumberOfFreeClusters, &TotalNumberOfClusters))
```

Comment:
Actually we can do better here, and use [GetDiskFreeSpaceExW](https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getdiskfreespaceexw) that removes the need to provide a root directory. (Suggested by @Extravert-ir )